### ISSUE TITLE
Initialize network weights in each K fold cross-validation

### ISSUE THREAD

**Issue by RezaKardoost** [ayman-elgharabawy/PNN#1]
I think that in the PNN.py when the k-fold cross-validation method is used, the network must be reinitialized for each cross-validation to create fresh w1 and w2.
The problem now is that the test samples in the nth cross-validation, are used as training samples in the n-1th cross-validation.
And this makes the model give us a better false answer in the last cross-validations because the test samples have already been trained in the model.

**Comment by ayman-elgharabawy** [comment]
The first point you are right, 

but for testing data , i divide the data to 80% training, 20% testing , then apply cros validation on 80% , then we have the model and apply the testing data.
you mean the validation not testing, right

**Comment by RezaKardoost** [comment]
Yes. I meant validation data. 

But then the question that comes to my mind is that when the k-fold method is used, we have finally trained k separated model. In other words, for each cross-validation, the network is initialized and trained.
Finally, which of the k models should be used to predict test(20%) data?

**Comment by ayman-elgharabawy** [comment]
Yes that is why i meant to add each fold to the same model, instead of initialize the network each time, because which model we will use for testing , that why i added each fold results to the same model.
